enterprise/coderd/prebuilds/metricscollector_test.go

```diff
@@ -1,73 +1,247 @@
+			shouldSetDesiredPrebuilds:        ptr.To(1.0),
+			shouldSetActualPrebuilds:         ptr.To(0.0),
+			shouldSetEligiblePrebuilds:       ptr.To(0.0),
+		},
+	}
+	for _, test := range tests {
+		for _, transition := range test.transitions {
+			for _, jobStatus := range test.jobStatuses {
+				for _, initiatorID := range test.initiatorIDs {
+					for _, ownerID := range test.ownerIDs {
+						t.Run(fmt.Sprintf("transition:%s/jobStatus:%s", transition, jobStatus), func(t *testing.T) {
+							t.Parallel()
+
+							logger := slogtest.Make(t, &slogtest.Options{IgnoreErrors: true})
+							t.Cleanup(func() {
+								if t.Failed() {
+									t.Logf("failed to run test: %s", test.name)
+									t.Logf("transition: %s", transition)
+									t.Logf("jobStatus: %s", jobStatus)
+									t.Logf("initiatorID: %s", initiatorID)
+									t.Logf("ownerID: %s", ownerID)
+								}
+							})
+							db, pubsub := dbtestutil.NewDB(t)
+							reconciler := prebuilds.NewStoreReconciler(db, pubsub, codersdk.PrebuildsConfig{}, logger, quartz.NewMock(t))
+							ctx := testutil.Context(t, testutil.WaitLong)
+
+							createdUsers := []uuid.UUID{prebuilds.OwnerID}
+							for _, user := range slices.Concat(test.ownerIDs, test.initiatorIDs) {
+								if !slices.Contains(createdUsers, user) {
+									dbgen.User(t, db, database.User{
+										ID: user,
+									})
+									createdUsers = append(createdUsers, user)
+								}
+							}
+
+							collector := prebuilds.NewMetricsCollector(db, logger, reconciler)
+							registry := prometheus.NewPedanticRegistry()
+							registry.Register(collector)
+
+							numTemplates := 2
+							for i := 0; i < numTemplates; i++ {
+								orgID, templateID := setupTestDBTemplate(t, db, ownerID)
+								templateVersionID := setupTestDBTemplateVersion(t, ctx, db, pubsub, orgID, ownerID, templateID)
+								preset := setupTestDBPreset(t, ctx, db, pubsub, templateVersionID, 1, uuid.New().String())
+								setupTestDBPrebuild(
+									t, ctx, db, pubsub,
+									transition, jobStatus, orgID, templateID, templateVersionID, preset.ID, initiatorID, ownerID,
+								)
+							}
+
+							metricsFamilies, err := registry.Gather()
+							require.NoError(t, err)
+
+							templates, err := db.GetTemplates(ctx)
+							require.NoError(t, err)
+							require.Equal(t, numTemplates, len(templates))
+
+							for _, template := range templates {
+								templateVersions, err := db.GetTemplateVersionsByTemplateID(ctx, database.GetTemplateVersionsByTemplateIDParams{
+									TemplateID: template.ID,
+								})
+								require.NoError(t, err)
+								require.Equal(t, 1, len(templateVersions))
+
+								presets, err := db.GetPresetsByTemplateVersionID(ctx, templateVersions[0].ID)
+								require.NoError(t, err)
+								require.Equal(t, 1, len(presets))
+
+								for _, preset := range presets {
+									checks := []metricCheck{
+										{"coderd_prebuilds_created", test.shouldIncrementPrebuildsCreated, true},
```

Comment:
Nit: can you add the labels here? It'd aid readability.

Reply:
I've moved these up into the test case definitions. Please check now?

Its not quite what you had in mind, but I do think its the better structure on balance of consideration.

Reply:
My point was more that the reader then had to find and memorise the order of fields in `metricCheck`, but this is cool for now.